integration-tests/splunk/src/main/java/org/apache/camel/quarkus/component/splunk/it/SplunkResource.java

```diff
@@ -0,0 +1,212 @@
+                .filter(e -> !e.getKey().startsWith("_"))
+                .collect(Collectors.toMap(
+                        Map.Entry::getKey,
+                        Map.Entry::getValue,
+                        (v1, v2) -> v1));
+
+        return result;
+    }
+
+    @Path("/startRealtimePolling")
+    @POST
+    public void startPolling(String search) {
+        before();
+        // use another thread for polling consumer to demonstrate that we can wait before
+        // the message is sent to the queue
+        Executors.newSingleThreadExecutor().execute(() -> {
+            String url = String.format(
+                    "splunk://realtime?scheme=http&port=%d&delay=3000&initEarliestTime=rt-10s&latestTime=RAW(rt+40s)&search="
+                            + search,
+                    port);
+            SplunkEvent body = consumerTemplate.receiveBody(url, SplunkEvent.class);
+            producerTemplate.sendBody("direct:realtimePolling", body);
+        });
+    }
+
+    @Path("/submit")
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.TEXT_PLAIN)
+    public Response submit(Map<String, String> message, @QueryParam("index") String index) throws Exception {
+        return post(message, "submit", index, null);
+    }
+
+    @Path("/stream")
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.TEXT_PLAIN)
+    public Response stream(Map<String, String> message, @QueryParam("index") String index) throws Exception {
+        return post(message, "stream", index, null);
+    }
+
+    @Path("/tcp")
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.TEXT_PLAIN)
+    public Response tcp(Map<String, String> message, @QueryParam("index") String index) throws Exception {
+        return post(message, "tcp", index, tcpPort);
+    }
+
+    private Response post(Map<String, String> message, String endpoint, String index, Integer tcpPort) throws Exception {
+        before();
+
+        SplunkEvent se = new SplunkEvent();
+        for (Map.Entry<String, String> e : message.entrySet()) {
+            se.addPair(e.getKey(), e.getValue());
+        }
+
+        String url = String.format(
+                "splunk:%s?scheme=http&port=%d&index=%s&sourceType=%s&source=%s",
+                endpoint, port, index, SOURCE_TYPE, SOURCE);
+        if (tcpPort != null) {
+            url = String.format(
+                    "splunk:%s?username=admin&password=changeit&scheme=http&port=%d&index=%s&sourceType=%s&source=%s&tcpReceiverLocalPort=%d&tcpReceiverPort=%d",
+                    endpoint, port, index, SOURCE_TYPE, SOURCE, LOCAL_TCP_PORT, tcpPort);
+        }
+        final String response = producerTemplate.requestBody(url, se, String.class);
+        return Response
+                .created(new URI("https://camel.apache.org/"))
+                .entity(response)
+                .build();
+    }
+
+    private void before() {
```

Comment:
Why do we need to set a configuration factory more than once ?

Reply:
Yes, you are right, I'll refactor code to configure component once, thanks.

Reply:
done